File: android/app/src/main/kotlin/pt/ualg/miaugenda/data/remote/AvailabilityApi.kt
package pt.ualg.miaugenda.data.remote

import pt.ualg.miaugenda.data.model.Availability
import pt.ualg.miaugenda.data.model.CreateAvailabilityBody
import retrofit2.Response
import retrofit2.http.*

interface AvailabilityApi {
    @GET("/availability")
    suspend fun getAvailabilities(
        @Query("startDate") startDate: String? = null,
        @Query("endDate") endDate: String? = null,
        @Query("userId") userId: Int? = null
    ): Response<List<Availability>>

    @POST("/availability")
    suspend fun createAvailability(@Body body: CreateAvailabilityBody): Response<Availability>

    @DELETE("/availability/{id}")
    suspend fun deleteAvailability(@Path("id") id: Int): Response<Unit>
}

File: desktop/src/api/availabilityApi.ts
import { apiClient } from "./apiClient";
import type { Availability, CreateAvailabilityBody } from "../models/availability";

export interface AvailabilityFilters {
  startDate?: string | null;
  endDate?: string | null;
  userId?: number | null;
}

export async function getAvailabilities(
  filters: AvailabilityFilters = {}
): Promise<Availability[]> {
  const response = await apiClient.get<Availability[]>("/availability", {
    params: {
      startDate: filters.startDate ?? undefined,
      endDate: filters.endDate ?? undefined,
      userId: filters.userId ?? undefined
    }
  });

  return response.data;
}

export async function createAvailability(
  body: CreateAvailabilityBody
): Promise<Availability> {
  const response = await apiClient.post<Availability>("/availability", body);
  return response.data;
}

export async function deleteAvailability(id: number): Promise<void> {
  await apiClient.delete(`/availability/${id}`);
}